<launch>

  <!-- Interface -->
  <!-- ========= -->

  <arg name="interface" default="keyboard"/>

  <!-- Use keyboard -->
  <group if="$(eval arg('interface') == 'keyboard')">
    <include file="$(find operator_node)/launch/keyboard.launch"/>
  </group>

  <!-- Use spacenav -->
  <group if="$(eval arg('interface') == 'spacenav')">

    <!-- Spacenav driver -->
    <node pkg="spacenav_node" type="spacenav_node" name="$(anon spacenav_node)" output="screen">
      <param name="zero_when_static" value="true"/>
      <param name="static_count_threshold" value="30"/>
    </node>

    <!-- Map spacenav to operator signal -->
    <node pkg="operator_node" name="joy2osig" type="scale_node.py">
      <remap from="joy" to="spacenav/joy"/>
      <rosparam param="config" file="$(find teleop)/configs/teleop_tf_demo_spacenav_config.yaml"/>
      <param name="start_on_init" type="bool" value="true"/>
    </node>
  </group>

  <!-- Teleoperation -->
  <!-- ============= -->

  <!-- Set eff start pose -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="tf_start_pose_in_world"
	args="0.5 0 0.5 0 0 0 world tf_start"/>

  <!-- Use operator signal to update a tf -->
  <node pkg="teleop" name="teleop_tf" type="teleop_tf_node.py">
    <rosparam param="config" file="$(find teleop)/configs/teleop_tf_demo_config.yaml"/>
    <param name="start_on_init" type="bool" value="true"/>
  </node>

  <!-- Visualization -->
  <!-- ============= -->

  <!-- Rviz -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find teleop)/rviz/teleop_tf_demo.rviz" />

</launch>
